index.js

```diff
@@ -77,6 +80,9 @@ export class Item {
     if (!list) {
       return false
     }
+    if (this === item) {
```

Comment:
these need a blank line before them, and a comment explaining why this check is needed